src/util/mod.rs

```diff
@@ -554,6 +555,16 @@ pub fn is_zero_duration(d: &Duration) -> bool {
     d.as_secs() == 0 && d.subsec_nanos() == 0
 }
 
+/// A fast hash function for u64.
+#[inline]
+pub fn hash_u64(i: u64) -> u64 {
+    let mut x = Wrapping(i);
+    x = (x ^ (x >> 30)) * Wrapping(0xbf58476d1ce4e5b9);
```

Comment:
What about using  the fnv hash instead? https://docs.rs/crate/fnv/1.0.6

Reply:
I think constructing a Hasher object each call may be costly because there is a hot path.